### PR TITLE
Change several api endpoints to be for internal use only

### LINKED ISSUE #7958: Move several API endpoints to internal only
From: https://docs.google.com/document/d/1KC07Rl_oD6jYUyYahv4UrS-GIuLWnr4biaWop826U1Y/edit

- [x] clinical-attributes/counts should be private (put in section clinical attribute counts) - Created a new controller
- [x] clinical-events move to private (prone to change) - Moved entire controller to internal
- [x] Move mutation-counts-by-position (maybe can be made private or organized by page?) - Create a new controller
- [x] /discrete-copy-number-counts (part of patient view page section?) - Create a new internal controller
- [x] Add description to internal page that they are prone to change
- [x] Add description to public endpoints that we intend to keep backwards compatibility (after 1.0 release)
- [x] Reference Genome Genes - Move to private
- [x] Resource Definitions - Move to private for now
- [x] Resource Data - Move to private for now

### PR DESCRIPTION
**Pull request by jagnathan**
Fix #7958 
- APIs have been moved to internal 
- Api Info has been updated for internal and public.

### CODE REVIEW

File: web/src/main/java/org/cbioportal/web/config/PublicApiTags.java [hunk 1]
```diff
@@ -7,6 +7,7 @@
     public static final String SAMPLES = "Samples";
     public static final String SAMPLE_LISTS = "Sample Lists";
     public static final String CLINICAL_ATTRIBUTES = "Clinical Attributes";
+    public static final String CLINICAL_ATTRIBUTES_COUNT = "Clinical Attributes Count";
```
Comment (inodb) [inline web/src/main/java/org/cbioportal/web/config/PublicApiTags.java:10]: @jagnathan should we maybe create a new class `InternalApiTags`?

**Review by inodb — APPROVED** [review]
LGTM! Thanks for doing this! I made one comment re api tags. Might be nice to change in this PR as well

File: web/src/main/java/org/cbioportal/web/MutationCountController.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
+package org.cbioportal.web;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.cbioportal.model.meta.MutationMeta;
+import org.cbioportal.model.Mutation;
+import org.cbioportal.model.MutationCountByPosition;
+import org.cbioportal.service.exception.MolecularProfileNotFoundException;
```
Comment (kalletlak) [inline web/src/main/java/org/cbioportal/web/MutationCountController.java:9]: Import not used

**Review by kalletlak — APPROVED** [review]
LGTM. One comment, cleanup unused imports in all the files that are updated/added.

**Comment by inodb** [comment]
@jagnathan there seem to be two consistent localdb failures. Is that just b/c of the frontend requiring some update?

**Pull request merged**